### PR TITLE
add solution, but with using context

### PR DESCRIPTION
**Pull request by ssaviak**
[DEMO LINK](https://ssaviak.github.io/react_pagination/)

### CODE REVIEW

**Review by vSainiuk — APPROVED** [review]
Well done!

File: src/components/Pagination/Pagination.tsx [hunk 1]
```diff
@@ -1 +1,59 @@
...
+            key={page}
+          >
+            <a
+              data-cy="pageLink"
+              className="page-link"
+              href={`#${page}`}
+              type="button"
+              onClick={() => {
+                setCurrentPage(page);
+              }}
```
Comment (DanilWeda) [inline src/components/Pagination/Pagination.tsx:39]: you can write a handler for it ( not critical )

File: src/components/Pagination/Pagination.tsx [hunk 2]
```diff
@@ -1 +1,59 @@
...
+
+  return (
+    <ul className="pagination">
+      <li className={cn('page-item', { disabled: currentPage === 1 })}>
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          aria-disabled="true"
+          onClick={() => setCurrentPage(prevPage => prevPage - 1)}
```
Comment (DanilWeda) [inline src/components/Pagination/Pagination.tsx:21]: you can write a handler for it ( not critical )

File: src/components/Pagination/Pagination.tsx [hunk 3]
```diff
@@ -1 +1,59 @@
...
+          </li>
+        ))
+      }
+      <li className={cn('page-item', { disabled: currentPage === pages })}>
+        <a
+          data-cy="nextLink"
+          className="page-link"
+          href="#next"
+          aria-disabled="false"
+          onClick={() => setCurrentPage(prevPage => prevPage + 1)}
```
Comment (DanilWeda) [inline src/components/Pagination/Pagination.tsx:52]: you can write a handler for it ( not critical )

File: src/PageContext/PageContext.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  currentPage: 1,
+  setTotal: () => {},
+  setCurrentPage: () => {},
+  setItemsPerPage: () => {},
+};
+
+export const PageContext
+  = React.createContext<PageContextValue>(initialValues);
+
+type Props = {
```
Comment (DanilWeda) [inline src/PageContext/PageContext.tsx:18]:
```suggestion
interface Props = {
```

**Review by DanilWeda — APPROVED** [review]
Great!
Approved. 
Many thanks!

**Pull request closed without merging**